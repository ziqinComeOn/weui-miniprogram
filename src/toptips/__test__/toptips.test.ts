import path from 'path'
import simulate from 'miniprogram-simulate'

describe('toptips', () => {
    const toptips = simulate.load(path.resolve(__dirname, '../toptips'), 'mp-toptips')

    test('basic', async () => {
        const comp = simulate.render(toptips, {
            extClass: 'top100',
            msg: '距离顶部100px',
            type: 'error',
            show: true
        })
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()
    })

    test('auto hide', async () => {
        const comp = simulate.render(
            simulate.load({
                usingComponents: {
                    'mp-toptips': toptips
                },
                template: `
                    <mp-toptips msg="500ms后关闭" type="success" delay="{{500}}" show="{{show}}" />
                `,
                data: {
                    show: false
                }
            })
        )
        comp.attach(document.createElement('parent-wrapper'))
        await simulate.sleep(0)
        comp.setData({ show: true })
        await simulate.sleep(0)
        expect(comp.toJSON()).toMatchSnapshot()

        await simulate.sleep(600)
        expect(comp.toJSON()).toMatchSnapshot()
    })
})
