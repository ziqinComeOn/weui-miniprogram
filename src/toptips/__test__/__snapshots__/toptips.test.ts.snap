// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toptips auto hide 1`] = `
<main>
  <mp-toptips>
    <wx-view
      class="weui-toptips weui-toptips_success  weui-toptips_show"
    >
      500ms后关闭
    </wx-view>
  </mp-toptips>
</main>
`;

exports[`toptips auto hide 2`] = `
<main>
  <mp-toptips>
    <wx-view
      class="weui-toptips weui-toptips_success  "
    >
      500ms后关闭
    </wx-view>
  </mp-toptips>
</main>
`;

exports[`toptips basic 1`] = `
<mp-toptips>
  <wx-view
    class="weui-toptips weui-toptips_error top100 weui-toptips_show"
  >
    距离顶部100px
  </wx-view>
</mp-toptips>
`;
